### PR TITLE
make second param of GeneratePublicApi truly optional, and align extension methods

### LINKED ISSUE #139: 10.0.0 changes make it impossible to call GeneratePublicApi with just an assembly argument
```c#
ApiGenerator.GeneratePublicApi(typeof(Foo).Assembly));
```

> The call is ambiguous between the following methods or properties: 'ApiGenerator.GeneratePublicApi(Assembly, ApiGeneratorOptions?)' and 'ApiGenerator.GeneratePublicApi(Assembly, Type[]?, bool, string[]?, string[]?)'

This worked in 9.x.

### PR DESCRIPTION
**Pull request by adamralph**
fixes #139

### CODE REVIEW

**Comment by sungam3r** [comment]
When obsolete overload will be removed it will be possible to return `= null` in the first extension method.

**Comment by adamralph** [comment]
@sungam3r I don't understand what you mean. Are you saying there is a problem with this change?

File: src/PublicApiGenerator/ApiGenerator.cs [hunk 1]
```diff
@@ -76,14 +76,14 @@ public static string GeneratePublicApi(this Type[] types, ApiGeneratorOptions? o
...
         /// <param name="options">The options to control the API output.</param>
         /// <returns>The API output.</returns>
-        public static string GeneratePublicApi(this Type type, ApiGeneratorOptions? options = null)
+        public static string GeneratePublicApi(this Type type, ApiGeneratorOptions? options)
         {
             (options ??= new ApiGeneratorOptions()).IncludeTypes = new Type[] { type };
             return type.Assembly.GeneratePublicApi(options);
         }
 
         [Obsolete("Use `GeneratePublicApi(this Assembly assembly, ApiGeneratorOptions? options = null)` instead. Will be removed in the next major.")]
```
Comment (danielmarbach) [inline src/PublicApiGenerator/ApiGenerator.cs]: Is this message still clear or should we remove the `= null`?

**Comment by sungam3r** [comment]
No problems.
image
I talk about that we can return `= null` for this method in next major version when obsolete method will be removed.

File: src/PublicApiGenerator/ApiGenerator.cs (hunk 1 above)
Comment (sungam3r) [reply src/PublicApiGenerator/ApiGenerator.cs]: To be consistent we should remove `= null`.

**Comment by adamralph** [comment]
@sungam3r thanks, I didn't realise that that method is obsolete. In that case, I don't think this PR is required. Consumers should switch to the other method, and in doing so, can switch extension method syntax which will make the second parameter optional.

**Comment by sungam3r** [comment]
The bug turned into a feature :). So we "motivate" the client to use the new extension method.

**Pull request closed without merging**